### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by huythangphan**

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -19,13 +19,16 @@
 BATCH_SIZE = 32
 NUM_CLASSES = 4
 
+# epochs
 EPOCHS=50
```
Comment (sherlockvn) [inline config.py]: epoch should be lied in above group.

File: dataframe.py [hunk 2]
```diff
@@ -62,19 +58,16 @@ def load_dataframe():
...
 def train_test(dataframe):
-    '''
-    '''
     X = list(dataframe["image"])
     y = list(dataframe["target"])
 
     return train_test_split(X, y, train_size=0.40, random_state=13)
 
 
+# pca model
```
Comment (sherlockvn) [inline dataframe.py:69]: pca model is not used in our code, right ? remove it

File: fit_model.py [hunk 3]
```diff
@@ -1,37 +1,39 @@
...
+      test_data=test_generator,
+      test_steps=total_test//BATCH_SIZE,
       steps_per_epoch=total_train//BATCH_SIZE,
       callbacks=callbacks_list
   )
 
-  nb_samples = validate_df.shape[0]
-  predict = model.predict(validation_generator, steps=np.ceil(nb_samples/BATCH_SIZE))
-  validate_df['pred'] = np.argmax(predict, axis=-1)
+  nb_samples = test_df.shape[0]
```
Comment (sherlockvn) [inline fit_model.py]: some comments here maybe better

File: generator.py [hunk 4]
```diff
@@ -1,20 +1,25 @@
-from config import IMG_DIR, IMAGE_SIZE, BATCH_SIZE
-from keras.preprocessing.image import ImageDataGenerator
 import random
 import numpy as np
+# configs
+from config import IMG_DIR, IMAGE_SIZE, BATCH_SIZE
+# keras api
+from keras.preprocessing.image import ImageDataGenerator
 
+# const
```
Comment (sherlockvn) [inline generator.py:8]: you should explain its use not its type

File: generator.py [hunk 5]
```diff
@@ -34,21 +39,21 @@ def generator(train_df, validate_df):
...
-      color_mode = 'rgb',
+      color_mode=COLOR_MODE,
       class_mode='categorical',
       batch_size=BATCH_SIZE
   )
 
-  validation_datagen = ImageDataGenerator(rescale=1./255)
-  validation_generator = validation_datagen.flow_from_dataframe(
-      validate_df, 
+  test_datagen = ImageDataGenerator(rescale=1./255)
```
Comment (sherlockvn) [inline generator.py]: some comments here may be better

File: open_cam.py [hunk 6]
```diff
@@ -1,71 +1,88 @@
...
 from imutils.video import VideoStream
+# keras api
+from tensorflow.python.keras.models import load_model
 from tensorflow.keras.applications.mobilenet_v2 import preprocess_input
 from tensorflow.keras.preprocessing.image import img_to_array
 
-
+# load model
 keras_model = load_model('model.h5')
+# facenet
```
Comment (sherlockvn) [inline open_cam.py]: comments its use not its name dude :)

File: open_cam.py [hunk 7]
```diff
@@ -1,71 +1,88 @@
...
 facenet = cv2.dnn.readNetFromCaffe(CAFFE_PROTO_FILE, MODEL_FILE)
```
Comment (sherlockvn) [inline open_cam.py:17]: for example, load predictive model from prototype and trained model file  caffenet which is for detecting human face

File: open_cam.py [hunk 8]
```diff
@@ -1,71 +1,88 @@
...
 
+# labels
```
Comment (sherlockvn) [inline open_cam.py:19]: maybe explains about the color and structure is better

File: training_model.py [hunk 9]
```diff
@@ -1,39 +1,39 @@
...
-
-
-
-
+print("[INFO] dataframe is loaded!")
 
+# train test
+X_train, X_test, y_train, y_test = train_test(dataframe) 
 
+# pca model
```
Comment (sherlockvn) [inline training_model.py:36]: this code is not running right ?
you can remove it

File: dataframe.py (hunk 2 above)
Comment (huythangphan) [reply dataframe.py:69]: it is used in training_model.py sir

File: generator.py (hunk 4 above)
Comment (huythangphan) [reply generator.py:8]: they are constant values so we just comment to understand what it is. their usage will be known where they are used

File: open_cam.py (hunk 6 above)
Comment (huythangphan) [reply open_cam.py]: :) no here is comment for definition not where it take action. only declaration

File: open_cam.py (hunk 8 above)
Comment (huythangphan) [reply open_cam.py:19]: here can easily understand sir

File: generator.py (hunk 4 above)
Comment (sherlockvn) [reply generator.py:8]: when we use uppercase word and underscore between, it is constant, so you dont have to comment like that

File: generator.py (hunk 5 above)
Comment (sherlockvn) [reply generator.py]: like why do you choose 1./255 ?

File: open_cam.py (hunk 6 above)
Comment (sherlockvn) [reply open_cam.py]: i think you can remove this comment, because it's not nescessary

File: open_cam.py (hunk 8 above)
Comment (sherlockvn) [reply open_cam.py:19]: oh really, so why you have to name variable is label ? (51, 153, 255) -> color ?

**Comment by sherlockvn** [comment]
change the README.md file also, for example, python3 open_cam.py is no way work because there is no open_cam.py file in current directory

**Comment by huythangphan** [comment]
@sherlockvn I have resolved all comments. please take a look at my PR again.
